src/pseudopeople/utilities.py

```diff
@@ -11,14 +12,23 @@ def get_randomness_stream(form: Form, seed: int) -> RandomnessStream:
     return RandomnessStream(form.value, lambda: pd.Timestamp("2020-04-01"), seed)
 
 
-def get_default_configuration() -> ConfigTree:
+def get_configuration(user_yaml_path: Union[Path, str] = None) -> ConfigTree:
+    """
+    Gets a noising configuration ConfigTree, optionally overridden by a user-provided YAML.
+
+    :param user_yaml_path: A path to the YAML file defining user overrides for the defaults
+    :return: a ConfigTree object of the noising configuration
+    """
     import pseudopeople
 
     default_config_layers = [
         "base",
+        "user",
     ]
-    noising_configuration = ConfigTree(layers=default_config_layers)
-    BASE_DIR = Path(pseudopeople.__file__).resolve().parent
-    yaml_path = BASE_DIR / "default_configuration.yaml"
-    noising_configuration.update(yaml_path, layer="base")
+    noising_configuration = ConfigTree(
+        data=Path(pseudopeople.__file__).resolve().parent / "default_configuration.yaml",
+        layers=default_config_layers,
+    )
+    if user_yaml_path:
+        noising_configuration.update(user_yaml_path, layer="user")
```

Comment:
So this is adding an additional layer to the config tree instead of generating a config tree with the usual layers we would expect with the user provided values?  Does that mean the user can override only certain config options and leave default values for others?

Reply:
I'm not sure what you mean by usual layers. This config tree will have two layers: base, and user. The base layer is populated by the default configuration. The user layer is populated by user provided configuration file. The user doesn't have to configure values for all keys - any keys they don't specify will use the value from the base layer.